lib/kazoo/partition.rb

```diff
@@ -16,7 +16,7 @@ def replication_factor
     end
 
     def preferred_leader
-      @replicas.first
+      @replicas.compact.first
```

Comment:
I think the problem lies somewhere else.

Kafka stores the replicas in order of preference. The first one is the preferred replica. E.g. `[1,2,3]`, in which 1 is the preferred one. 

In this case, `1` doesn't get set because that broker is down. However, that doesn't change the fact that `1` is the preferred replica. The real issue is: why does it show up as `nil` when it is down?